reps/templates/python/{{cookiecutter.__project_slug}}/pyproject.toml

```diff
@@ -36,6 +37,10 @@ target-version = "py{{cookiecutter.__min_tox_python_version}}"
 [tool.ruff.isort]
 known-first-party = ["{{cookiecutter.__package_name}}"]
 
+[tool.pyright]
+include = ["src/{{cookiecutter.__package_name}}"]
+reportUnknownParameterType = "error"
```

Comment:
There are other places we can enforce types, but this catches the most valuable ones (function signatures). I'd be open to being more strict, but also don't want to cause too much friction.